### PR TITLE
Encode OAuth headers

### PR DESCRIPTION
**Pull request by charmander**
Essentially, this doesn’t work at all right now.

### CODE REVIEW

File: weasyl/oauth2.py [hunk 1]
```diff
@@ -66,21 +74,15 @@ def authorize_post_(request):
                                     form.username, form.password, bool(form.remember_me),
                                     bool(form.not_me)))
     credentials['userid'] = userid
-    headers, body, status = server.create_authorization_response(
+    response = server.create_authorization_response(
```
Comment (hyena) [inline weasyl/oauth2.py]: I'd prefer to reserve the name `response` for Pyramid response objects only. Everywhere else in the code-base that's what it refers to, so avoiding breaking that here was a part of least surprise.

File: weasyl/oauth2.py [hunk 2]
```diff
@@ -66,21 +74,15 @@ def authorize_post_(request):
...
 
 
 @disallow_api
 def token_(request):
-    headers, body, status = server.create_token_response(*extract_params())
-    for k, v in headers.iteritems():
-        request.set_header_on_response(k, v)
-    request.set_status_on_response("%s Status" % (status,))
-    return Response(body)
+    response = server.create_token_response(*extract_params(request))
```
Comment (hyena) [inline weasyl/oauth2.py]: Here as well.

File: weasyl/oauth2.py [hunk 3]
```diff
@@ -6,12 +6,20 @@
...
+            body=body,
+            status_code=status,
+            headers={k.encode('utf-8'): v.encode('utf-8') for k, v in headers.iteritems()},
+        )
 
 
 def extract_params(request):
-    headers = http.get_headers(request.environ)
-    return request.path, request.method, request.params, headers
+    return request.path, request.method, request.params, request.headers
```
Comment (hyena) [inline weasyl/oauth2.py]: Should https://github.com/Weasyl/weasyl/blob/9dea21db3c8ddb22dbf804366834df9d2feb7f5a/weasyl/middleware.py#L350 change as well?

**Comment by hyena** [comment]
Inheritance, eh. Alright.

File: weasyl/oauth2.py (hunk 3 above)
Comment (charmander) [inline weasyl/oauth2.py]: Probably. For another pull request.

**Pull request merged**